internal/loader/configuration_setting_loader.go

```diff
@@ -743,3 +759,51 @@ func MergeSecret(secret map[string]corev1.Secret, newSecret map[string]corev1.Se
 
 	return nil
 }
+
+func createCorrelationContextHeader(provider acpv1.AzureAppConfigurationProvider, clientManager ClientManager) http.Header {
+	header := http.Header{}
+	output := make([]string, 0)
+
+	if provider.Spec.Configuration.Refresh != nil {
+		output = append(output, "RefreshesKeyValue")
+	}
+
+	if provider.Spec.Secret != nil {
+		output = append(output, "UsesKeyVault")
+
+		if provider.Spec.Secret.Refresh != nil &&
+			provider.Spec.Secret.Refresh.Enabled {
+			output = append(output, "RefreshesKeyVault")
+		}
+	}
+
+	if provider.Spec.FeatureFlag != nil {
+		output = append(output, "UsesFeatureFlag")
+
+		if provider.Spec.FeatureFlag.Refresh != nil &&
+			provider.Spec.FeatureFlag.Refresh.Enabled {
+			output = append(output, "RefreshesFeatureFlag")
```

Comment:
Feature flag by nature should be refreshed. This is not useful to collect.

The purpose of `RefreshesKeyVault` is to understand how many stores use appconfig for automatic secret rotation.